### PR TITLE
[AKS] `az aks create`: Add edge zone support

### PR DESCRIPTION
**Pull request by rheaparekh**
**Description**
Adding edge zone support in the `az aks create` command

- Add the extended location parameter to create AKS in the edge zones

- Add a test case for the cluster creation in edge zone

**Testing Guide**
Added the unit test `test_aks_create_edge_zone` as well as manually tested by creating AKS at the edge zones on a test subscription

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by zhoxing-ms — APPROVED** [review]

**Comment by andyliuliming** [comment]
@zhoxing-ms could you please help wait for the approval from AKS team before merge the PR?
this feature is not working now.

and it broke our testing pipeline.

**Pull request merged**
